### ISSUE TITLE
Terminated Account Behavior

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#8]
In GitLab by @kdjstudios on Oct 21, 2016, 19:45

Keener:

a. You may have seen the question we had before about cancelled accounts: To our horror, we realized that once you terminate an account, the leftover usage charges and especially late charges, do not get applied.
In order to create late charges, you must keep the account active, which gives a totally wrong number of active accounts in the system. AND you have to now manually remove the Recurring, Periodic, and Holiday Fees. 
b. Also, if you then give that DID number to a new client, (which is equal to the resource ID), the usage from the new client may be applied to the old client, or Vice Versa, or Both, which is extremely messy.
We wish for a “Collections Office” where terminated accounts can sit and rack up late charges until paid or written off to Bad Debt.

. Once a termed account has a zero balance, they should be “parked” in a “Termed Account” section/office where they are no longer counted as accounts, and where we can run reports on “termed” accounts only.
a. When I open up SA billing, it shows me this below. I am pretty sure that this is the Total number of customers/accounts in the system, not the total “ACTIVE” customers/accounts.

Tim says: The accounts should first go into an 'inactive' state in which late charges accrue but no usage or base rate items. Only after an account has been satisfied (paid or written off) should it be terminated.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Aug 26, 2019, 12:11

@tim.traylor 

Keeping the 'what to charge' in #9; we should be able to focus this ticket on how a terminated account can be re organized in order to separate out from certain reports.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2019, 09:46

Closing ticket, this is being addressed in #1585

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 27, 2019, 09:46

closed
